src/test/ui/invalid-self-argument/trait-fn.rs

```diff
@@ -0,0 +1,11 @@
+struct Foo {}
+
+impl Foo {
+    fn c(foo: u32, self) {}
+    //~^ ERROR unexpected `self` argument in function
+    //~| NOTE `self` is only valid as the first argument of a trait function
```

Comment:
Since this is a function in an inherent impl, it would be nice if the error could emphasize here that the problem is that it must be the *first* argument.

Reply:
will fix

Reply:
It's not a big deal, but still would be nice.

But I am not knowledgeable in the relevant APIs either.^^